src/scheduler/fifo.c

```diff
@@ -1242,17 +1242,12 @@ run_job(int pbs_sd, resource_resv *rjob, char *execvnode, int throughput, schd_e
 					snprintf(logbuf, MAX_LOG_SIZE, "Job will run for duration=%s", timebuf);
 					schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, rjob->name, logbuf);
 				}
-				if (throughput)
-					rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-				else
-					rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 			}
-		} else {
-			if (throughput)
-				rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-			else
-				rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 		}
+		if (rc > 0 && throughput)
+			rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
+		else
```

Comment:
oops, ya, my bad, it could be a : if (rc > 0) { if (throughout) else } block

Between, I'd asked this before, shouldn't we be bailing out if update_job_attr fails? The if (rc) block at the end doesn't handle return values of update_job_attr() if it fails.